src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -686,10 +695,12 @@ func (r *ClusterReconciler) reportStatus(
 }
 
 func statusShouldBeUpdated(
+	ctx context.Context,
 	status *redpandav1alpha1.ClusterStatus,
 	nodeList *redpandav1alpha1.NodesList,
 	sts *resources.StatefulSetResource,
 ) bool {
+	currentVersion, versionErr := sts.CurrentVersion(ctx)
```

Comment:
nit, non blocking: generally it should not fail, but this call can be externalized and `currentVersion` passed as input to simplify the logic